### DISCUSSION TITLE
Vehicle detection problem

### DISCUSSION THREAD

**Discussion by mltraore** [AlexeyAB/darknet#7837] [Q&A]
when detecting vehicles, is it better to categorize vehicles before detection or to identify them all by labeling them as a single class?

**Comment by woosal1337** [comment]
> when detecting vehicles, is it better to categorize vehicles before detection or to identify them all by labeling them as a single class?

What do you mean? How many different vehicle types are you wishing to detect? How different are they from each other?

**Reply by mltraore** [reply]
I want to detect vehicles and people. Vehicles such as motorcycles, trams and trains are also included in the category of vehicles together with cars. For now, I have gathered all the vehicles under one label. The name of this label is "vehicle". People have a separate label called "person". When detecting these vehicles, is it better to label all of them as vehicles or label all of them as separate vehicles? Such as car, motorcycle etc.

**Comment by Aniruddha7** [comment]
If you try to include all different vehicle classes under one class, it
might affect training since the features of different types of vehicles
differ. I think it would be better to separate out trucks, cars and
motorcycles.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
